lib/public/components/PublicFeedSourceViewer.js

```diff
@@ -1,21 +1,55 @@
-import React, {Component, PropTypes} from 'react'
+// @flow
+
+import React, {Component} from 'react'
 import { Grid, Row, Col, Table, Panel, Glyphicon } from 'react-bootstrap'
 import { Link } from 'react-router'
 
-import PublicPage from './PublicPage'
+import * as feedsActions from '../../manager/actions/feeds'
+import * as versionsActions from '../../manager/actions/versions'
 import ActiveFeedVersionNavigator from '../../manager/containers/ActiveFeedVersionNavigator'
+import PublicPage from './PublicPage'
 
-export default class PublicFeedSourceViewer extends Component {
-  static propTypes = {
-    feedSource: PropTypes.object,
-    routeParams: PropTypes.object,
-    user: PropTypes.object,
+import type {Props as ContainerProps} from '../containers/ActivePublicFeedSourceViewer'
+import type {Feed, Project} from '../../types'
+import type {ManagerUserState} from '../../types/reducers'
 
-    onComponentMount: PropTypes.func
-  }
+type Props = ContainerProps & {
+  feedSource: Feed,
+  fetchFeedSourceAndProject: typeof feedsActions.fetchFeedSourceAndProject,
+  fetchFeedVersions: typeof versionsActions.fetchFeedVersions,
+  project: Project,
+  runFetchFeed: typeof feedsActions.runFetchFeed,
+  updateFeedSource: typeof feedsActions.updateFeedSource,
+  uploadFeed: typeof versionsActions.uploadFeed,
+  user: ManagerUserState
+}
+
+export default class PublicFeedSourceViewer extends Component<Props> {
   componentWillMount () {
-    this.props.onComponentMount(this.props)
+    const {
+      feedSource,
+      fetchFeedSourceAndProject,
+      fetchFeedVersions,
+      routeParams,
+      user
+    } = this.props
+    let unsecured = true
+    if (user.profile !== null) {
+      unsecured = false
+    }
+    if (!feedSource && routeParams.feedSourceId) {
+      fetchFeedSourceAndProject(routeParams.feedSourceId, unsecured)
+        // $FlowFixMe action is wrapped in dispatch which returns a promise
+        .then((feedSource) => {
+          console.log('fetch versions')
+          fetchFeedVersions(feedSource, unsecured)
+        })
+    // $FlowFixMe FIXME does this variable still exist in the feedSource type?
```

Comment:
I think it is actually called `feedVersions`, which is the list of versions is added to its parent feed source object in the projects reducer under `RECEIVE_FEEDVERSIONS`. Perhaps we could handle this better, but yea `FeedSource#versions` doesn't appear to be a thing.